src/python/pants/util/osutil.py

```diff
@@ -11,6 +11,14 @@
 logger = logging.getLogger(__name__)
 
 
+# We use `sched_getaffinity()` to get the number of cores available to the process, rather than
+# the raw number of cores. This ensures that containers report their # of cores, rather than
```

Comment:
This wording lies: ~ s/ensures/sometimes helps/ ... As noted in my experiments, this does not catch restrictions via Docker --cpus for example which can be used to throttle %utilization without pinning to a cpu like --cpuset-cpus does.